cmis_mail/wizards/mail_compose_message.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2022 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import base64
+import logging
+import mimetypes
+import os
+from io import BytesIO
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class MailComposeMessage(models.TransientModel):
+    _inherit = "mail.compose.message"
+
+    is_save_in_cmis_enabled = fields.Boolean(
+        string="Save attachments in CMIS",
+        default=False,
+    )
+    allowed_cmis_folder_field_ids = fields.Many2many(
+        comodel_name="ir.model.fields",
+        readonly=True,
+    )
+    is_multiple_cmis_fields = fields.Boolean(
+        readonly=True,
+    )
+    cmis_folder_field_id = fields.Many2one(
+        string="CMIS Folder",
+        comodel_name="ir.model.fields",
+        domain="[('id', 'in', allowed_cmis_folder_field_ids.ids)]",
+    )
+    cmis_duplicate_handler = fields.Selection(
+        selection=[
+            ("use_existing", "Use existing"),
+            ("error", "Raise exception"),
+            ("new_version", "Create a new version"),
+            ("increment", "Rename as file(X).pdf"),
+        ],
+        string="Duplicate strategy",
+        default="increment",
+    )
+
+    @api.model
+    def default_get(self, fields):  # pylint:disable=redefined-outer-name
+        res = super().default_get(fields)
+        related_model = res.get("model")
+        if not related_model:
+            return res
+        cmis_fields = (
+            self.env["ir.model.fields"]
+            .sudo()
+            .search([("model", "=", related_model), ("ttype", "=", "cmis_folder")])
+        )
+        if cmis_fields:
+            res.update(
+                {
+                    "allowed_cmis_folder_field_ids": [(6, 0, cmis_fields.ids)],
+                    "cmis_folder_field_id": cmis_fields[0].id,
+                    "is_multiple_cmis_fields": len(cmis_fields) > 1,
+                }
+            )
+        return res
+
+    def _get_cmis_parent_folder(self):
+        self.ensure_one()
+        field_name = self.cmis_folder_field_id.name
```

Comment:
Access to ir.model.fields that will cause an error if not sudo
```suggestion
        field_name = self.cmis_folder_field_id.sudo().name
```